keps/sig-windows/1122-windows-csi-support/README.md

```diff
@@ -768,12 +777,366 @@ Downgrading the version of csi-proxy.exe to one that is not supported by all ins
+  - Components depending on the feature gate:
+- [x] Other
+  - Describe the mechanism:
+  CSI-proxy is a binary that can be deployed (or running as a service) directly on each Windows node during node start up time. For example, in gce, CSI-proxy
+  binary is downloaded and running as Windows service in node [startup script](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/windows/k8s-node-setup.psm1#L424)
+  An environment variable ENABLE_CSI_PROXY is implemented to disable/enable CSI-proxy installation.
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+  If csi-proxy binary is deployed as a binary or windows service during node start up time, enabling this feature will require reprovisioning of a node.
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+It depends on cloud provider implementation. For example in GCE, environment variable ENABLE_CSI_PROXY is implemented to disable/enable CSI-proxy installation.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Nothing
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+-->
+No
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+It is up to cloud provider to have their own plan. For example, GKE will use version control to rollout, upgrade and rollback this feature.
+Please also see (#upgrade--downgrade-strategy)
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+It is up to the cloud provider to tie CSI-proxy version to cluster version. When clusters rollout and rollback, the CSI-proxy should be udpated accordingly.
```

Comment:
Is it a requirement to tie the csi-proxy version to a Kubernetes version?

Reply:
I agree technically there is no dependency between csi proxy and kubernetes versions (besides 1.18). However, for csi drivers that are not cloud-specific, their support matrix would be simpler if they could indicate a Kubernetes version and Kubernetes distros adhere to our recommendations. Otherwise one k8s distro on 1.20 may support csi-proxy v1beta1, but another distro may only support csi-proxy v1beta2, etc. This becomes especially complicated when a driver also has to take into consideration the csi sidecar version dependencies.